src/sync.cpp

```diff
@@ -2190,33 +2201,172 @@ void UnifiedSync::changedConfigState(bool notifyApp)
 }
 
 Syncs::Syncs(MegaClient& mc)
-    : mClient(mc)
+  : mClient(mc)
 {
     mHeartBeatMonitor.reset(new BackupMonitor(&mClient));
 }
 
+JSONSyncConfigDB* Syncs::syncConfigDB()
+{
+    // Have we already created the database?
+    if (mSyncConfigDB)
+    {
+        // Yep, return a reference to the caller.
+        return mSyncConfigDB.get();
+    }
+
+    // Is the client using a database?
+    if (!mClient.dbaccess)
+    {
+        // Nope and we need it for the configuration path.
+        return nullptr;
+    }
+
+    // Can we get our hands on an IO context?
+    if (!syncConfigIOContext())
+    {
+        // We need it if we want to write the DB to disk.
+        return nullptr;
+    }
+
+    // Where the database will be stored.
+    auto dbPath = mClient.dbaccess->rootPath();
+
+    // Create the database.
+    mSyncConfigDB.reset(new JSONSyncConfigDB(dbPath));
+
+    return mSyncConfigDB.get();
+}
+
+bool Syncs::syncConfigDBDirty()
+{
+    return mSyncConfigDB && mSyncConfigDB->dirty();
+}
+
+error Syncs::syncConfigDBFlush()
+{
+    if (!syncConfigDBDirty())
+    {
+        return API_OK;
+    }
+
+    LOG_verbose << "Attempting to flush internal config database.";
+
+    auto result = mSyncConfigDB->write(*mSyncConfigIOContext);
+
+    if (result != API_OK)
+    {
+        LOG_err << "Couldn't flush internal config database to disk: "
+                << result;
+    }
+    else
+    {
+        LOG_verbose << "Internal config database flushed to disk.";
+    }
+
+    return result;
+}
+
+error Syncs::syncConfigDBLoad()
+{
+    LOG_verbose << "Attempting to load internal sync configs from disk.";
+
+    auto result = API_EAGAIN;
+
+    // Can we get our hands on the internal sync config database?
+    if (auto* db = syncConfigDB())
+    {
+        // Make sure we have a suitable IO context.
+        if (auto* ioContext = syncConfigIOContext())
+        {
+            // Try and read the database from disk.
+            result = db->read(*ioContext);
+        }
+
+        if (result == API_OK || result == API_ENOENT)
```

Comment:
I think it can be, it the call `result = db->read(*ioContext);` finds that there is no file on disk to load.   I think that test is there in ordrer to log that 0 syncs were loaded, and have the return code indicate that there's no actual problem, as that is a perfectly legitimate scenario.

Reply:
you are corect, I jumped into the wrong::read call when I was checking that. Apologies